src/lib/components/Text.tsx

```diff
@@ -9,11 +9,13 @@ import getRegularStyles from '../utils/getRegularStyles'
 import { ITextProps } from '../..'
 import getTextChildren from '../utils/getTextChildren'
 import { flattenStyle } from '../utils/flattenStyle'
+import { ifDirective } from '../utils/directives'
 
 const Text: React.FunctionComponent<ITextProps & IConsumerInjectedProps> = ({
   children,
   theme,
   style,
+  nIf,
```

Comment:
Maybe instead of destructuring directive props here you can just pass "rest" to the renderWithDirective function (when it's been refactored to the util folder (if you decide do do that)). Then you can destructure the props in that function instead. Less clutter in this file :)